pkg/tfvars/gcp/gcp.go

```diff
@@ -41,6 +45,9 @@ func TFVars(auth Auth, masterConfigs []*gcpprovider.GCPMachineProviderSpec, imag
 		VolumeSize:              masterConfig.Disks[0].SizeGb,
 		ImageURI:                imageURI,
 		PublicZoneName:          publicZoneName,
+		ClusterNetwork:          masterConfig.NetworkInterfaces[0].Network,
```

Comment:
Oh I see. Thanks for clearing that up. I'm glad it is not the serious problem I thought it was. 

I just pushed a fix which continues with this approach and fixes the issue by passing the PreexistingNetwork  bool in to tfvars. I think this might be good because we're doing a little less logic in TF, and continuing to use machine objects as you suggested. Let me know what you think.